### PR TITLE
Deprecate legacy doc

### PR DESCRIPTION
**Pull request by frkv**
Documenting deprecation of legacy Mbed TLS crypto toolbox APIs

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -95,6 +95,9 @@ Security
 
 |no_changes_yet_note|
 
+Deprecated:
+* Deprecated legacy Mbed TLS crypto toolbox APIs that are enabled when :kconfig:option:`CONFIG_NORDIC_SECURITY_BACKEND` Kconfig is set. Please use :kconfig:option:`CONFIG_NRF_SECURITY` and PSA crypto APIs instead.
+
```
Comment (mia-ko) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:100]:
```suggestion

* Deprecated:

  * Legacy Mbed TLS crypto toolbox APIs that are enabled when the :kconfig:option:`CONFIG_NORDIC_SECURITY_BACKEND` Kconfig option is set.
    Use the nRF Security (enabled with the :kconfig:option:`CONFIG_NRF_SECURITY` Kconfig option) and PSA crypto APIs instead.

```

Comment (mia-ko) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:100]: This changelog entry should be removed from your PR, and added as a change request to PR: https://github.com/nrfconnect/sdk-nrf/pull/18192

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/security/nrf_security/doc/backend_config.rst [hunk 2]
```diff
@@ -10,6 +10,10 @@ Legacy configurations and supported features
 
 This section covers the configurations available when using :ref:`legacy_crypto_support`.
 
+.. caution::
+   Legacy crypto toolbox APIs is marked as deprecated in nRF Connect SDK 2.8.0, and will removed in a future version.
+   It is not recommended to rely on the configurations presented here or the APIs they configure in any new designs.
```
Comment (mia-ko) [inline doc/nrf/libraries/security/nrf_security/doc/backend_config.rst:15]:
```suggestion
.. caution::
   Legacy crypto toolbox APIs is marked as deprecated in the |NCS| version 2.8.0, and will be removed in a future version.
   It is not recommended to use the legacy crypto toolbox APIs and the related configurations in any new designs.
```

File: doc/nrf/libraries/security/nrf_security/doc/configuration.rst [hunk 3]
```diff
@@ -19,25 +19,23 @@ PSA crypto support is included by default when you enable nRF Security through t
...
+To enable the legacy crypto support mode of nRF Security, set the :kconfig:option:`CONFIG_NORDIC_SECURITY_BACKEND` Kconfig option along with additional configuration options, as described in :ref:`nrf_security_legacy_config`.
 The legacy crypto support allows backwards compatibility for software that requires usage of Mbed TLS crypto toolbox functions prefixed with ``mbedtls_``.
 
+.. caution::
+   Mbed TLS Legacy crypto toolbox APIs are marked as deprecated in nRF Connect SDK version 2.8.0, and will be removed in a future version. It is not recommended to make use of crypto toolbox functions prefixed with ``mbedtls_`` for any new designs. Please use equivalent functionality from PSA crypto APIs, instead.
+   Setting the :kconfig:option:`CONFIG_NORDIC_SECURITY_BACKEND` Kconfig option will enable the :kconfig:option:`MBEDTLS_LEGACY_CRYPTO_C` Kconfig option which will show a deprecation warning in the build output. Please use ::kconfig:option:`CONFIG_NRF_SECURITY` instead.
```
Comment (mia-ko) [inline doc/nrf/libraries/security/nrf_security/doc/configuration.rst:32]:
```suggestion
.. caution::
   Mbed TLS legacy crypto toolbox APIs are marked as deprecated in the |NCS| version 2.8.0, and will be removed in a future version.
   It is not recommended to use crypto toolbox functions prefixed with ``mbedtls_`` for any new designs.
   Use the equivalent functionality from PSA crypto APIs instead.
   
   Setting the Kconfig option :kconfig:option:`CONFIG_NORDIC_SECURITY_BACKEND` will also enable the Kconfig option :kconfig:option:`MBEDTLS_LEGACY_CRYPTO_C`, which will show a deprecation warning in the build output.
   Use the Kconfig option :kconfig:option:`CONFIG_NRF_SECURITY` instead.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by endre-nordic — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (divipillai) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:100]: I have added this change here: https://github.com/nrfconnect/sdk-nrf/compare/41f93f0b887ebeb166cbf4c7d24cd398642d841c..195d67075aa037c6b435111d26a223b35ded9ce1 . Please remove this entry from this PR.

File: doc/nrf/libraries/security/nrf_security/doc/backend_config.rst (hunk 2 above)
Comment (melwee01) [reply doc/nrf/libraries/security/nrf_security/doc/backend_config.rst:15]: Subject-verb disagreement with 'APIs is'. APIs are?

Comment (mia-ko) [reply doc/nrf/libraries/security/nrf_security/doc/backend_config.rst:15]: Argh, good catch. I read API LOL

**Review by divipillai — APPROVED** [review]

File: doc/nrf/libraries/security/nrf_security/doc/backend_config.rst [hunk 4]
```diff
@@ -10,6 +10,10 @@ Legacy configurations and supported features
 
 This section covers the configurations available when using :ref:`legacy_crypto_support`.
 
+.. caution::
+   Legacy crypto toolbox APIs is marked as deprecated in the |NCS| version 2.8.0, and will be removed in a future version.
```
Comment (mia-ko) [inline doc/nrf/libraries/security/nrf_security/doc/backend_config.rst:14]:
```suggestion
   Legacy crypto toolbox APIs are marked as deprecated in the |NCS| version 2.8.0, and will be removed in a future version.
```

**Pull request merged**